packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +199,153 @@ class CupertinoAlertDialog extends StatelessWidget {
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return new ClipRRect(
+      borderRadius: const BorderRadius.only(
+        bottomLeft: const Radius.circular(12.0),
+        bottomRight: const Radius.circular(12.0),
+      ),
+      child: actionSection,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        width: _kCupertinoDialogWidth,
+        padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+        child: new CustomMultiChildLayout(
+          delegate: new _CupertinoAlertDialogLayoutDelegate(
+            actionsCount: actions.length,
+          ),
+          children: <Widget>[
+            new LayoutId(
+              id: _AlertDialogSection.blur,
+              child: _buildBlurBackground(),
+            ),
+            new LayoutId(
+              id: _AlertDialogSection.content,
+              child: _buildContent(),
+            ),
+            new LayoutId(
+              id: _AlertDialogSection.actions,
+              child: _buildActions(),
+            ),
+          ],
         ),
       ),
     );
   }
 }
 
+enum _AlertDialogSection {
+  blur,
+  content,
+  actions,
+}
+
+class _CupertinoAlertDialogLayoutDelegate extends MultiChildLayoutDelegate {
+
+  final int actionsCount;
+
+  _CupertinoAlertDialogLayoutDelegate({
+    this.actionsCount,
+  });
+
+  @override
+  void performLayout(Size size) {
+    // TODO: @mattcarroll, handle 2 buttons that need to stack due to long text
+    double minActionSpace = 0.0;
+    if (actionsCount > 0 && actionsCount <= 2) {
+      minActionSpace = _kButtonHeight;
+    } else {
+      minActionSpace = 1.5 * _kButtonHeight;
+    }
+
+    // Size alert dialog content.
+    final Size maxContentSize = new Size(size.width, size.height - minActionSpace);
+    final Size contentSize = layoutChild(
+        _AlertDialogSection.content,
+        new BoxConstraints.loose(maxContentSize),
+    );
+
+    // Size alert dialog actions.
+    final Size maxActionSize = new Size(
+        size.width,
+        size.height - contentSize.height,
+    );
+    final Size actionsSize = layoutChild(
+        _AlertDialogSection.actions,
+        new BoxConstraints.loose(maxActionSize),
+    );
+
+    // Calculate overall dialog height.
+    final double dialogHeight = contentSize.height + actionsSize.height;
+
+    // Size blur background.
+    final Size backgroundSize = new Size(size.width, dialogHeight);
+    layoutChild(
+      _AlertDialogSection.blur,
+      new BoxConstraints.loose(backgroundSize),
+    );
+
+    // Layout the blur, content, and the actions.
+    final double dialogTop = (size.height - dialogHeight) / 2;
+    positionChild(
+      _AlertDialogSection.blur,
```

Comment:
Ya, I might be missing something but from reading this, I still think we don't need this

Reply:
spoke offline. the blur is needed in the custom layout because the parent layout is the full height of the screen.  therefore moving the blur to the parent would blur most of the screen.